app/build.gradle

```diff
@@ -10,35 +12,57 @@ android {
 
 tasks.register('conanProfile', Copy) {
     from "conanprofile.txt"
-    into "build/"
+    into project.layout.buildDirectory
+    filter(ReplaceTokens, tokens: ["NDK_PATH": android.ndkDirectory.toString()])
+}
 
-    doLast {
-        def file = file("build/conanprofile.txt")
-        def content = file.text
-        content = content.replace("<NDK_PATH>", android.ndkDirectory.toString())
-        file.write(content)
+abstract class ConanInstallTask extends Exec {
+    @Input
+    abstract Property<String> getArch()
+
+    // Cannot read project.layout.buildDirectory when configuration-cache is enabled
+    @Input
+    abstract Property<String> getArchBuildDirPath()
+
+    @OutputDirectory
+    final Provider<Directory> outputDirectory = archBuildDirPath.map { new File(it) }
+
+    @Override
+    protected void exec() {
+        commandLine(
+            "conan", "install", ".",
+            "--output-folder=" + archBuildDirPath.get(),
+            "--build=missing",
+            "--profile:host=build/conanprofile.txt",
+            "-s", "arch=" + arch.get(),
+            "-s", "build_type=Release",
+            "-s", "&:build_type=RelWithDebInfo",
+            "-s", "odrcore/*:build_type=RelWithDebInfo",
+        )
+        super.exec()
     }
 }
 
-tasks.register('conanInstall') {
-    dependsOn(tasks.named("conanProfile"))
-    doFirst {
-        ["armv7", "armv8", "x86", "x86_64"].each { String arch ->
-            exec {
-                commandLine(
-                    "conan", "install", ".",
-                    "--output-folder=build/conan/" + arch,
-                    "--build=missing",
-                    "--profile:host=build/conanprofile.txt",
-                    "-s", "arch=" + arch,
-                    "-s", "build_type=Release",
-                    "-s", "&:build_type=RelWithDebInfo",
-                    "-s", "odrcore/*:build_type=RelWithDebInfo",
-                )
-            }
+["armv8", "armv7", "x86", "x86_64"].each { architecture ->
```

Comment:
should we put this in a list somewhere?

Reply:
Architecture list? Well, it is going to change sometime in the future, once we drop armv7 support, but I doubt that's anytime soon

Reply:
I just saw that we have the same list twice so we might put this in a variable or somewhere. But nothing crucial

Reply:
True, arched are defined in two places. However, once our plugin is published, most of this will be removed, so I'll just keep this as is and merge, because we don't know how soon it will be published